kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/ControllerUtils.java

```diff
@@ -212,15 +212,17 @@ public static String marshal(String marshallingFormat, Object entity) {
                 break;
             }
         }
-
+        if (entity instanceof ServerTemplate) {
+            return marshaller.marshall(((ServerTemplate) entity).cloneServerTemplate());
```

Comment:
So there is still a tiny chance of a concurrent modification if this thread and the delete thread iterate over the `serverInstances` colletion, isn't it? Just if I get it right...

Reply:
I think yes, last version code I added LOCK for it, but I find if remove it, the test also can pass, So I change it.